packages/editor-ui/src/api/helpers.ts

```diff
@@ -0,0 +1,75 @@
+import axios, { AxiosRequestConfig, Method } from 'axios';
+import {
+	IDataObject,
+} from 'n8n-workflow';
+import {
+	IRestApiContext,
+} from '../Interface';
+
+
+class ResponseError extends Error {
+	// The HTTP status code of response
+	httpStatusCode?: number;
+
+	// The error code in the resonse
+	errorCode?: number;
+
+	// The stack trace of the server
+	serverStackTrace?: string;
+
+	/**
+	 * Creates an instance of ResponseError.
+	 * @param {string} message The error message
+	 * @param {number} [errorCode] The error code which can be used by frontend to identify the actual error
+	 * @param {number} [httpStatusCode] The HTTP status code the response should have
+	 * @param {string} [stack] The stack trace
+	 * @memberof ResponseError
+	 */
+	constructor (message: string, errorCode?: number, httpStatusCode?: number, stack?: string) {
+		super(message);
+		this.name = 'ResponseError';
+
+		if (errorCode) {
+			this.errorCode = errorCode;
+		}
+		if (httpStatusCode) {
+			this.httpStatusCode = httpStatusCode;
+		}
+		if (stack) {
+			this.serverStackTrace = stack;
+		}
+	}
+}
+
+export default async function makeRestApiRequest(context: IRestApiContext, method: Method, endpoint: string, data?: IDataObject): Promise<any> { // tslint:disable-line:no-any
+	try {
+		const {baseURL, sessionid} = context;
+		const options: AxiosRequestConfig = {
+			method,
+			url: endpoint,
+			baseURL,
+			headers: {
+				sessionid,
```

Comment:
this I think has to stay like this, because the expected header https://github.com/n8n-io/n8n/blob/c83c05456db008a47d3b3d1b7af89eabeb0c380b/packages/cli/src/GenericHelpers.ts#L57